### DISCUSSION TITLE
Preparing a notebook on "what is GDP and why does it matter?"

### DISCUSSION THREAD

**Discussion by jstac** [QuantEcon/reading-group-2024#10] [General]
@mbek0605 @Jiarui-ZH @pgrosser1 

Thanks for the work you did preparing the notebook(s) on components of GDP and correlations between GDP and other variables.

It would be great if you could cherry pick the best, most informative figures and put them in a single notebook.  We'll then share that with Tom, get feedback, and consider adding some of them to the live site.

(Please cut the discussion surrounding the figures to only what's most important --- the data source, a particularly interesting observation, etc.)

(I'm sorry to say that you won't be credited as a co-author of any lecture that we might produce from your figures.  Instead, you will get a small amount of credit on https://intro.quantecon.org/about.html.  It's a bad deal for RAs but we try to help you in other ways.)

I suggest the following sections

### What is GDP?

- @Jiarui-ZH looked into this and perhaps he could summarize that discussion --- perhaps just a few sentences, an example, etc.
- @mbek0605 has started work on some nice colored bar graphs that show what expenditure (one view of GDP) consists of (investment, consumption, etc.) in the US and how it's changed over time.  Perhaps @mbek0605 and @Jiarui-ZH could collaborate to finish up those figures.

### GDP and human welfare

- @pgrosser1 has made lots of nice figures and I suggest that we include at least one figure for each topic (GDP vs longevity, GDP vs HDI, etc.)

### Negative externalities

- GDP growth is also associated with environmental destruction.  To illustrate, it would be great if @Jiarui-ZH and @mbek0605 could collaborate to finish up the GDP vs global temperature figure, hopefully going back further into the past.
- Anything else?

**Comment by jstac** [comment]
@Jiarui-ZH and @mbek0605 : 

Data for global temperatures going further into the past: https://www.newscientist.com/article/2167127-why-the-hockey-stick-graph-will-always-be-climate-sciences-icon/

For GDP, we already have suitable data in the long run growth lecture.

**Comment by jstac** [comment]
Thanks for submitting the notebook @mbek0605 @pgrosser1 @Jiarui-ZH .  It's very nicely written and constructed.

To finish it off, it would be great if you could do the following:

- [ ]  Please use the section titles I suggested above (e.g., section 2 should be "GDP and human welfare")
- [ ]  The figure on nominal vs real GDP should start from the base year.
- [ ]  Regarding @mbek0605 's figures on components of GDP, let's drop the trends and bring back the bar graph that shows components of GDP as a share of GDP for the US over time.  The components of GDP are `GDP = C + I + G + NX` where `NX = X - M` --- see @Jiarui-ZH 's discussion.  Hence if we take all four components and divide them by GDP, we should get shares that sum to 1.  That bar graph is the one I'm hoping for.
- [ ]  @pgrosser1 , can we add a plot of GDP per capita vs self-reported happiness? 
- [ ]  @Jiarui-ZH and @mbek0605 I would love to see a version of the temperature anomoly vs GDP graph that goes back further in time.  I know it's not easy but would you mind to try?  @mmcky and @HumphreyYang might have suggestions on data sources.
- [ ]  In this negative externality section, I'd also love to see a similar plot but with a measure of biodiversity instead of temperature.  Anyone have any thoughts on how we might do this?

For future edits you need to make a branch, edit the md file using the jupyterlab-myst extension to Jupyter Lab, put these changes in a branch, and submit a pull request.  I realize this will be hard for those who are new to git but @mmcky and @HumphreyYang are here to help, and those are great skills to learn. 

(It will become clear why we store myst md files instead of ipynb files over time.  Basically, GitHub is for managing code and a myst md file is like code while an ipynb file is not really human readable.)

**Reply by HumphreyYang** [reply]
Hi team,

Please find the PAGES2k ensamble prediction of 2000 years of temperature data below:

https://figshare.com/articles/dataset/Reconstruction_ensemble_median_and_95_range/8143094

source: https://www.nature.com/articles/s41561-019-0400-0

**Reply by jstac** [reply]
Nice work @HumphreyYang

**Reply by mmcky** [reply]
Hi team,

You can also find an estimate of global long run GDP:

https://ourworldindata.org/grapher/global-gdp-over-the-long-run

which is derived from the Angus Maddison dataset as a special WB project.

**Reply by mbek0605** [reply]
Hello, 

I am reaching out with several questions: 

1. Regarding the bar graph, please let me know which one (first or second) is better. I am open to any other way 
![image](https://github.com/QuantEcon/reading-group-2024/assets/157935657/dd493ceb-04b7-4752-ac5d-cff2feece3fa)
![image](https://github.com/QuantEcon/reading-group-2024/assets/157935657/dcb16ed4-a378-4788-8742-ef05782103f2)

2. I tried to reference the PAGES2k ensemble prediction of 2000 years of temperature data, but I am getting an error: 
![image](https://github.com/QuantEcon/reading-group-2024/assets/157935657/d438bc8a-3b51-4cdb-8ae1-abbb18932388)

Please let me know how else I can improve my section. I can happily improve before our meeting despite my late outreach. Apologies.

**Reply by mmcky** [reply]
@mbek0605 I am looking at the graphs and the `negative` values for `trade deficit` make interpreting the shares confusing as you would expect the bar chart (shares) to sum to 1. 

What about you try the following:

Compute all positive values in the GDP identity (as shares) which would include `Exports` as an item (rather than a `X-M`) and you put `Exports` at the bottom of the chart. Then you could have `Imports` as negative values off the bottom of the chart (and `X-M`) could be an overlay line plot to show the deficit (due to Importing more than the US Exports). Then the bar chart above the `x-axis` will always sum to one, and the line plot would give you an indication of the size of the `trade deficit`

Here is a little sketch of what I am suggesting. 

![image](https://github.com/QuantEcon/reading-group-2024/assets/8263752/32906173-4c19-4d58-b09f-8949164df691)

What do you think?

**Reply by HumphreyYang** [reply]
> I tried to reference the PAGES2k ensemble prediction of 2000 years of temperature data, but I am getting an error

I find the webpage works well on my end: [Full_ensemble_median_and 95pct_range.txt](https://github.com/QuantEcon/reading-group-2024/files/14445935/Full_ensemble_median_and.95pct_range.txt)

There might be some issues on the server side blocking users' access. Please refer to @Jiarui-ZH 's comment below if you still have issues. RoG is also a great journal on earth science.

**Reply by mbek0605** [reply]
Sounds good! I will try my best to replicate that chart. Thank you very much for providing the visual aid!

**Reply by mbek0605** [reply]
In case I fail to make a pull request, I am attaching an image of the chart: 
![Untitled](https://github.com/QuantEcon/reading-group-2024/assets/157935657/c2362953-f8ba-447b-a5a9-ea77395b4ac2)

**Comment by Jiarui-ZH** [comment]
I had a similar problem with the PAGES2K data, instead I digitalised the
findings of https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2003RG000143
which gave an rough estimation of the temperature prior to 1850.

**Reply by mbek0605** [reply]
Jiarui was very helpful and kind in helping me integrate this data into my graph! Many thanks!

**Comment by Jiarui-ZH** [comment]
I was able to construct the graph using the PAGES2k data set which is better data set than the RoG data. Here is the result: , let me know what you think. 
![image](https://github.com/QuantEcon/reading-group-2024/assets/155980488/c8efb849-acd1-423d-b015-69e47ec01189)
